### PR TITLE
Add brunsli

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@conda-forge/help-c-cpp i don't get why the `.lib` file isn't being included in the install.

Any help understanding that would be greatly appreciated!

File: recipes/brunsli/use_conda_forge_shared_libraries.patch [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ set(WASM_MODULES brunslicodec-wasm brunslidec-wasm brunslienc-wasm)
+@@ -133,11 +148,24 @@ endif()  # BRUNSLI_EMSCRIPTEN
+ # Installation
+ if(NOT BRUNSLI_EMSCRIPTEN)
+   install(
+-    TARGETS brunslidec-c brunslienc-c
+-    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
++    TARGETS brunslidec-c
++    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
++    ARCHIVE DESTINATION "${CMAKE_INSTALL_BINDIR}"
```
Comment (nehaljwani) [inline recipes/brunsli/use_conda_forge_shared_libraries.patch:78]:
```suggestion
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
```

File: recipes/brunsli/use_conda_forge_shared_libraries.patch [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
++    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
++  )
++
++  install(
++    TARGETS brunslienc-c
++    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
++    ARCHIVE DESTINATION "${CMAKE_INSTALL_BINDIR}"
```
Comment (nehaljwani) [inline recipes/brunsli/use_conda_forge_shared_libraries.patch:85]:
```suggestion
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
```

File: recipes/brunsli/use_conda_forge_shared_libraries.patch (hunk 1 above)
Comment (hmaarrfk) [reply recipes/brunsli/use_conda_forge_shared_libraries.patch]: I diagree with this suggestion
```
ARCHIVE

    Target artifacts of this kind include:

        Static libraries (except on macOS when marked as FRAMEWORK, see below);

        DLL import libraries (on all Windows-based systems including Cygwin; they have extension .lib, in contrast to the .dll libraries that go to RUNTIME);
```

**Comment by hmaarrfk** [comment]
maybe you are right.... i'm juts confused on this one ;)

File: recipes/brunsli/use_conda_forge_shared_libraries.patch (hunk 1 above)
Comment (nehaljwani) [reply recipes/brunsli/use_conda_forge_shared_libraries.patch]: The defaults are sane for each platform, so I generally don't specify these unless I really wish to override. I would recommend watching this excellent talk: https://www.youtube.com/watch?v=m0DwB4OvDXk (can skip to 28:02 for this specific topic)

**Comment by hmaarrfk** [comment]
The problem seems to be that all lib files are omitted
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=176705&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=26d9c833-f81c-502c-87bc-0cb0be44f838&l=1119

**Comment by nehaljwani** [comment]
I think when you do:
```cmake
target_link_libraries(brunslienc-c
  brotlienc
  brotlicommon
)
```
`brunslienc-c` in itself has nothing to export, that's why there is no import lib for it. 

IMO, their CMakeLists.txt is correct and you don't need to split all. They simply want to provide a decode shared library and an encode shared library. The `-static` library is deliberately intermediate (named incorrectly, I'll give you that) and not supposed to be installed separately.

**Comment by hmaarrfk** [comment]
but the decode shared library, doesn't have an associated `.lib` is that correct?

**Comment by hmaarrfk** [comment]
On windows, somehow, it is trying to install brotli???

image

I've simply modified the cmake file to use our shared version of brotli instead of their versionned one:

Should i build with the `-static` version instead?

I thought that conda-forge didn't want static libraries, and that they would be removed pretty soon from the default packages

image

**Comment by nehaljwani** [comment]
Your current CMake changes are almost correct [ideally it should have used find_package() for brotli and then used the imported targets]

The problem right now is that brusli doesn't export symbols on Windows properly.
They have all the required macros defined (but aren't using them anywhere in headers):
https://github.com/google/brunsli/blob/master/c/common/port.h#L262-L290

One way to get around that would be to use `-DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=1` in the Windows build, but that is usually frowned upon.

**Comment by hmaarrfk** [comment]
Alright, well in that case, it seems like an upstream issue.

I reported it and will let them deal with it.

For now skipping the windows builds!

Thanks for the help!!!

**Comment by scopatz** [comment]
Thanks @hmaarrfk!

**Pull request merged**